Sources/TokamakCore/Gestures/GestureMask.swift

```diff
@@ -0,0 +1,94 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Szymon on 30/7/2023.
+//
+
+import Foundation
+
+/// Options that control how adding a gesture to a view affects other gestures recognized by the view and its subviews.
+@frozen public struct GestureMask: Equatable, ExpressibleByArrayLiteral, OptionSet, Sendable {
+    public typealias RawValue = Int8
+    public var rawValue: Int8
+
+    // MARK: - OptionSet
+
+    public init(rawValue: Int8) {
+        self.rawValue = rawValue
+    }
+
+    // MARK: - Equatable
+
+    public static func == (lhs: GestureMask, rhs: GestureMask) -> Bool {
+        return lhs.rawValue == rhs.rawValue
+    }
+
+    // MARK: - ExpressibleByArrayLiteral
+
+    /// Creates a gesture mask from an array of gesture options.
+    ///
+    /// - Parameter elements: An array of `GestureMask` elements.
+    public init(arrayLiteral elements: GestureMask...) {
+        self.rawValue = elements.reduce(0) { $0 | $1.rawValue }
+    }
+
+    // MARK: - SetAlgebra
+
+    static var allZeros: GestureMask {
+        return GestureMask(rawValue: 0)
+    }
+
+    static func | (lhs: GestureMask, rhs: GestureMask) -> GestureMask {
+        return GestureMask(rawValue: lhs.rawValue | rhs.rawValue)
+    }
+
+    static func & (lhs: GestureMask, rhs: GestureMask) -> GestureMask {
+        return GestureMask(rawValue: lhs.rawValue & rhs.rawValue)
+    }
+
+    static prefix func ~ (x: GestureMask) -> GestureMask {
+        return GestureMask(rawValue: ~x.rawValue)
+    }
```

Comment:
Are you sure about this? 
When I remove this I have bunch of error such as 
```
Referencing operator function '|' on 'BinaryInteger' requires that 'GestureMask' conform to 'BinaryInteger'
```

Reply:
Instead of using bitwise operators you can use [`SetAlgebra`](https://developer.apple.com/documentation/swift/setalgebra) methods. `GestureMask` in SwiftUI does not implement those operators either.

For example, you should be able to replace `.gesture | .subviews` with `[.gesture, .subviews]`

Reply:
I see, thanks for getting back to me @carson-katri 
Fixed!

Reply:
run all the checks, more files than actually edited did show up with changes. 
Had to commit them as it is a pre-commit action now :P